inc/common.php

```diff
@@ -565,10 +565,15 @@ function ml($id = '', $more = '', $direct = true, $sep = '&amp;', $abs = false)
         $more = str_replace(',', $sep, $more);
     }
 
-    if($abs) {
-        $xlink = DOKU_URL;
-    } else {
-        $xlink = DOKU_BASE;
+    if ($conf[cdn] && $direct) {
```

Comment:
Splitbrain said "a patch", so I interpreted that as a core patch and not a plugin. Cf. https://github.com/splitbrain/dokuwiki/issues/1020#issuecomment-73473979